### PR TITLE
[Shop][Cart] Update cart after channel switching  

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
CartBundle cannot be aware of channels, alternative implementation must be in core.

File: features/cart/shopping_cart/viewing_cart_summary_in_many_channels.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
+@shopping_cart
+Feature: Viewing a cart summary in many channels
+    In order to see details about my order in selected channel
+    As a Visitor
+    I want to be able to see my cart summary in selected channel
```
Comment (NeverResponse) [inline features/cart/shopping_cart/viewing_cart_summary_in_many_channels.feature]: `I want to see different carts in different channels`? or something alike

**Comment by pjedrzejewski** [comment]
Thank you Ania! :)

File: src/Sylius/Bundle/CartBundle/EventListener/SessionCartSubscriber.php [hunk 2]
```diff
@@ -75,8 +75,11 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
-        if (null !== $cart && null !== $cart->getId()) {
-            $request->getSession()->set($this->sessionKeyName, $cart->getId());
+        if (null !== $cart && null !== $cart->getId() && null !== $cart->getChannel()) {
+            $request->getSession()->set(
+                sprintf('%s.%s', $this->sessionKeyName, $cart->getChannel()->getCode()),
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CartBundle/EventListener/SessionCartSubscriber.php]: I think that cart is not aware of channel in `CartBundle`, this change should be reverted.

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/CartRepository.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+use Sylius\Bundle\CartBundle\Doctrine\ORM\CartRepository as BaseCartRepository;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\CartRepositoryInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class CartRepository extends BaseCartRepository implements CartRepositoryInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/CartRepository.php]: I hate this `Order` -> `Cart` -> `Order` design, but as far as I'm familiar with this concept in `Core` we should have `Order`, so this should be renamed to `OrderRepository`... 😕

**Pull request merged**